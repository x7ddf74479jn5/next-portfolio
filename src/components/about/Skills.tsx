import React from "react";
import HeadLine from "src/components/about/HeadLine";
import { WrapInSection } from "src/components/layouts/Section";
import ServiceItem from "src/components/projects/ServiceItem";
import styles from "src/styles/components/about/Skill.module.scss";

import { services } from "../../contents/services";
import type { Skill } from "../../contents/skills";
import { skills } from "../../contents/skills";

type Props = {
  skill: Skill;
};

export const SkillItem: React.VFC<Props> = ({ skill }) => {
  return (
    <div className={styles.wrapper}>
      <h3 className={styles.head}>{skill.head}</h3>
      <div className={styles.data}>{skill.data}</div>
    </div>
  );
};

const Skills = () => {
  return (
    <WrapInSection id="skills">
      <HeadLine>Skills</HeadLine>
      <p>技術と開発物は以下の通りです。</p>
      <div className={styles.gridList}>
        {services.map((service, index) => {
          return <ServiceItem service={service} key={index} />;
        })}
      </div>
      <ul className={styles.gridTable}>
        {skills.map((skill) => {
          return (
            <li key={skill.id}>
              <SkillItem skill={skill} />
            </li>
          );
        })}
      </ul>
    </WrapInSection>
  );
};

export default Skills;
