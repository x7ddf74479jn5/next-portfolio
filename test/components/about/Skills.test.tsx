import { render } from "@testing-library/react";
import Skills from "src/components/about/Skills";

describe("Skills", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<Skills />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("rendered correctly", () => {
    const { container, getByRole } = render(<Skills />);

    const sectionElement = container.firstElementChild;
    expect(sectionElement).toHaveAttribute("id", "skills");

    const headerElement = getByRole("heading", { level: 2 });
    expect(headerElement).toHaveTextContent("Skills");

    const skillList = getByRole("list");
    expect(skillList.childElementCount).toBeGreaterThan(0);
  });
});
