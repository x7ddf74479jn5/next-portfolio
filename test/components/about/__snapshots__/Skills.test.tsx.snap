// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skills matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="section"
    id="skills"
  >
    <div
      class="sectionWrapIn"
    >
      <h2
        class="headlineAbout"
      >
        Skills
      </h2>
      <p>
        技術と開発物は以下の通りです。
      </p>
      <div
        class="gridList"
      >
        <article
          class="gridListItemFloated"
        >
          <h3>
            WEBサイト制作
          </h3>
          <small>
            Website Production
          </small>
          <img
            alt="Website Production"
            class="mediaThumb"
            height="288"
            src="/img/services/website-production.jpeg"
            width="512"
          />
          <p>
            コーポレートサイト、個人ポートフォリオ、ランディングページなどWebサイト制作全般を承ります。検索流入やSNS流入を強化する「SEOを意識したサイト」を作成できます。
          </p>
        </article>
        <article
          class="gridListItemFloated"
        >
          <h3>
            WordPress
          </h3>
          <small>
            WordPress
          </small>
          <img
            alt="WordPress"
            class="mediaThumb"
            height="288"
            src="/img/services/wordpress-icon.png"
            width="512"
          />
          <p>
            WordPress導入サポートやカスタマイズ、その他お困り事のご相談にも対応いたします。
          </p>
        </article>
        <article
          class="gridListItemFloated"
        >
          <h3>
            チャットアプリ開発
          </h3>
          <small>
            Chat APP
          </small>
          <img
            alt="Chat APP"
            class="mediaThumb"
            height="288"
            src="/img/services/web-app.jpeg"
            width="512"
          />
          <p>
            リアルタイム性が重視されるSNSアプリやチャットアプリなどを開発できます。これまで開発してきたWebアプリの一部を制作事例として公開しています。
          </p>
        </article>
      </div>
      <ul
        class="gridTable"
      >
        <li>
          <div
            class="wrapper"
          >
            <h3
              class="head"
            >
              マークアップ
            </h3>
            <div
              class="data"
            >
              HTML Living Standard, CSS3, Sass, CSS in JS
            </div>
          </div>
        </li>
        <li>
          <div
            class="wrapper"
          >
            <h3
              class="head"
            >
              フロントエンド
            </h3>
            <div
              class="data"
            >
              JavaScript, TypeScript, webpack, Babel, Jest, Storybook
            </div>
          </div>
        </li>
        <li>
          <div
            class="wrapper"
          >
            <h3
              class="head"
            >
              バックエンド
            </h3>
            <div
              class="data"
            >
              Firebase, Amplify
            </div>
          </div>
        </li>
        <li>
          <div
            class="wrapper"
          >
            <h3
              class="head"
            >
              フレームワーク・ライブラリ
            </h3>
            <div
              class="data"
            >
              React, Redux, Next.js, Tailwind CSS, MUI, Chakra UI
            </div>
          </div>
        </li>
        <li>
          <div
            class="wrapper"
          >
            <h3
              class="head"
            >
              その他ツール
            </h3>
            <div
              class="data"
            >
              Git, npm, Dockerなど
            </div>
          </div>
        </li>
      </ul>
    </div>
  </section>
</DocumentFragment>
`;
